pkg/networkservice/core/trace/client.go

```diff
@@ -16,76 +14,55 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// Package trace provides wrappers for tracing around a networkservice.NetworkServiceClient
 package trace
 
 import (
 	"context"
 
+	"github.com/sirupsen/logrus"
+
 	"github.com/golang/protobuf/ptypes/empty"
 	"google.golang.org/grpc"
 
 	"github.com/networkservicemesh/api/pkg/api/networkservice"
 
-	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
-	"github.com/networkservicemesh/sdk/pkg/tools/typeutils"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/trace/traceconcise"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/trace/traceverbose"
 )
 
-type beginTraceClient struct {
+type traceClient struct {
 	traced networkservice.NetworkServiceClient
-}
 
-type endTraceClient struct{}
+	verbose networkservice.NetworkServiceClient
+	concise networkservice.NetworkServiceClient
+}
 
 // NewNetworkServiceClient - wraps tracing around the supplied networkservice.NetworkServiceClient
 func NewNetworkServiceClient(traced networkservice.NetworkServiceClient) networkservice.NetworkServiceClient {
-	return next.NewNetworkServiceClient(
-		&beginTraceClient{traced: traced},
-		&endTraceClient{},
-	)
-}
-
-func (t *beginTraceClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*networkservice.Connection, error) {
-	// Create a new logger
-	operation := typeutils.GetFuncName(t.traced, methodNameRequest)
-	ctx, finish := withLog(ctx, operation, methodNameRequest, request.GetConnection().GetId())
-	defer finish()
-
-	logRequest(ctx, request, "request")
-	// Actually call the next
-	rv, err := t.traced.Request(ctx, request, opts...)
-	if err != nil {
-		return nil, logError(ctx, err, operation)
+	return &traceClient{
+		traced:  traced,
+		verbose: traceverbose.NewNetworkServiceClient(traced),
+		concise: traceconcise.NewNetworkServiceClient(traced),
 	}
-	logResponse(ctx, rv, "request")
-	return rv, err
 }
 
-func (t *beginTraceClient) Close(ctx context.Context, conn *networkservice.Connection, opts ...grpc.CallOption) (*empty.Empty, error) {
-	// Create a new logger
-	operation := typeutils.GetFuncName(t.traced, methodNameClose)
-	ctx, finish := withLog(ctx, operation, methodNameClose, conn.GetId())
-	defer finish()
-
-	logRequest(ctx, conn, "close")
-	rv, err := t.traced.Close(ctx, conn, opts...)
-	if err != nil {
-		return nil, logError(ctx, err, operation)
+func (t *traceClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*networkservice.Connection, error) {
+	switch logrus.GetLevel() {
+	case logrus.TraceLevel:
+		return t.verbose.Request(ctx, request, opts...)
+	case logrus.InfoLevel, logrus.DebugLevel:
```

Comment:
Hmm.. What about other levels?  E.g. Warn?

What if we follow logic loglevel less than tracelevel? 
yes - concise
no - verbose


Feel free to correct me if I miss something

Reply:
In the case of Warn (or Error), traces will not be displayed in any form. This will work like `next`.

Reply:
I think we need to always inject some tracing because without it we won't get connection id in logs from chain elements.